astropy/io/ascii/core.py

```diff
@@ -326,6 +326,10 @@ def get_lines(self, table, newline=None):
         lines : list
             List of lines
         """
+        if isinstance(table, os.PathLike):
+            # normalize path objects to string
+            table = os.fspath(table)
+
         try:
             if hasattr(table, "read") or (
```

Comment:
```suggestion
        try:
            if hasattr(table, "read") or isinstance(table, os.PathLike) or (
```
as it seems the only point of failure was actually the unsupported `'PosixPath' + 'str'`, this seems a less intrusive change – in particular as in the spirit of #14893 `get_readable_fileobj` should always support pathlike objects.